scanner.go

```diff
@@ -319,11 +348,19 @@ func (f *fetcher) shouldClose(resp *pb.ScanResponse, region hrpc.RegionInfo) boo
 
 	// Check to see if this region is the last we should scan because:
 	// (1) it's the last region
-	if len(region.StopKey()) == 0 {
+	if len(region.StopKey()) == 0 && !f.rpc.Reversed() {
+		return true
+	}
+	if f.rpc.Reversed() && len(region.StartKey()) == 0 {
 		return true
 	}
 	// (3) because its stop_key is greater than or equal to the stop_key of this scanner,
 	// provided that (2) we're not trying to scan until the end of the table.
+	if !f.rpc.Reversed() {
+		return len(f.rpc.StopRow()) != 0 && // (2)
+			bytes.Compare(f.rpc.StopRow(), region.StopKey()) <= 0 // (3)
+	}
+	//  Reversed Scanner
 	return len(f.rpc.StopRow()) != 0 && // (2)
-		bytes.Compare(f.rpc.StopRow(), region.StopKey()) <= 0 // (3)
+		bytes.Compare(f.rpc.StopRow(), region.StartKey()) >= 0 // (3)}
```

Comment:
we shouldn't stop if StopRow() of the scan is equal to StartKey() of the region because it's inclusive.

Reply:
actually nevermind, StopRow isn't inclusive.